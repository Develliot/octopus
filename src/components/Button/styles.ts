import styled from 'styled-components';

export const ButtonStyled = styled.button`
    position: relative;
    font-size: ${({
        theme: {
            fontSizes: { extraExtraLarge },
        },
    }) => extraExtraLarge};
    padding: ${({
        theme: {
            spacing: { small, large },
        },
    }) => `${small} ${large}`};
    font-weight: 800;
    transition: color 0.3s ease, background-color 0.3s ease;
    background-color: ${({
        color,
        theme: {
            colors,
            colors: { pink },
        },
    }) => (color ? colors[`${color}`] : pink)};
    border-width: 0.12rem;
    border-style: solid;
    border-radius: 0.5rem;
    border-color: ${({
        color,
        theme: {
            colors,
            colors: { pink },
        },
    }) => (color ? colors[`${color}`] : pink)};
    color: ${({
        color,
        theme: {
            colors: { white, darkBlue },
        },
    }) => {
        switch (color) {
            case 'white':
                return darkBlue;
            case 'lightBlue':
                return darkBlue;
            case 'blue':
                return white;
            case 'pink':
                return white;
            case 'darkGrey':
                return white;
            case undefined:
                return white;
            default:
                return white;
        }
    }};

    width: 100%;

    @media screen and (min-width: ${({
            theme: {
                breakpoints: { small },
            },
        }) => small}) {
        width: auto;
    }

    &:hover,
    &:focus {
        background-color: ${({
            color,
            theme: {
                colors: { white },
            },
        }) => {
            switch (color) {
                case 'white':
                    return `rgba(0, 0, 0, 0.4)`;
                case 'lightBlue':
                    return `rgba(0, 0, 0, 0.4)`;
                case 'blue':
                    return white;
                case 'pink':
                    return white;
                case 'darkGrey':
                    return white;
                case undefined:
                    return white;
                default:
                    return white;
            }
        }};
        color: ${({
            color,
            theme: {
                colors,
                colors: { pink },
            },
        }) => (color ? colors[`${color}`] : pink)};
    }

    &::after {
        content: '';
        display: ${({ isShiny }: { isShiny: boolean }) =>
            isShiny ? `block` : `none`};
        position: absolute;
        top: 0.05%;
        left: 0.05%;
        width: calc(100% - 0.1%);
        height: 50%;
        border-radius: 0.5rem;
        background: linear-gradient(
            rgba(255, 255, 255, 0.5),
            rgba(255, 255, 255, 0.2)
        );
    }
`;
