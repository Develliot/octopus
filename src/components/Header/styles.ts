import styled from 'styled-components';
const LogoSVG = require('src/assets/svg/logo.svg');
const MenuSVG = require('src/assets/svg/menu.svg');

export const HeaderWrapper = styled.header`
    position: fixed;
    top: 0;
    left: 0;
    right: 0;
    margin: 0 auto;
    width: 100vw;
    transition: margin ease 0.2s, opacity ease 0.3s;
    margin-top: ${({ visible }: { visible: boolean }) =>
        visible ? `0` : `-100px`};
    opacity: ${({ visible }: { visible: boolean }) => (visible ? `1` : `0`)};
    z-index: 10;
    border-top: ${({
        theme: {
            colors: { pink },
        },
    }) => `solid 0.001rem ${pink}`};
    background-color: ${({
        theme: {
            colors: { darkBlue },
        },
    }) => darkBlue};
`;

export const HeaderInner = styled.div`
    box-sizing: border-box;
    margin: 0 auto;
    background-color: ${({
        theme: {
            colors: { darkBlue },
        },
    }) => darkBlue};
    padding: ${({
        theme: {
            spacing: { small },
            globalPadding,
        },
    }) => `${small} ${globalPadding}`};
    max-width: ${({
        theme: {
            breakpoints: { large },
        },
    }) => large};
    margin: 0 auto;
    border-top: ${({
        theme: {
            colors: { pink },
        },
    }) => `solid 0.001rem ${pink}`};
`;

export const Links = styled.ul`
    max-width: ${({
        theme: {
            breakpoints: { large },
        },
    }) => large};

    display: flex;
    flex-direction: column;
    list-style-type: none;
    justify-content: flex-start;

    overflow: hidden;
    transition: height 0.3s ease;
    height: ${({ menuOpen }: { menuOpen: boolean }) =>
        menuOpen ? '100vh' : '0'};

    @media screen and (min-width: ${({
            theme: {
                breakpoints: { small },
            },
        }) => small}) {
        flex-direction: row;
        list-style-type: none;
        height: auto;
        padding: 0;
    }
`;

export const LogoSVGStyled = styled(LogoSVG)`
    fill: white;
    height: 2rem;
    width: auto;

    @media screen and (min-width: ${({
            theme: {
                breakpoints: { small },
            },
        }) => small}) {
        height: 3rem;
    }
`;

export const MenuIconStyled = styled(MenuSVG)`
    fill: ${({
        theme: {
            colors: { white },
        },
    }) => white};
    height: 100%;
    width: 100%;
    transition: fill 0.3s ease;
`;

export const MenuButtonStyled = styled.button`
    height: 2rem;
    width: 2rem;
    background: transparent;
    border: none;
    padding: 0;

    &:hover,
    &:focus {
        ${MenuIconStyled} {
            fill: ${({
                theme: {
                    colors: { pink },
                },
            }) => pink};
        }
    }

    @media screen and (min-width: ${({
            theme: {
                breakpoints: { small },
            },
        }) => small}) {
        display: none;
    }
`;

export const ResponsiveSpacer = styled.div`
    transition: height 0.3s ease;
    height: ${({ menuOpen }: { menuOpen: boolean }) =>
        menuOpen ? '1em' : '0'};
    @media screen and (min-width: ${({
            theme: {
                breakpoints: { small },
            },
        }) => small}) {
        height: 1em;
    }
`;
