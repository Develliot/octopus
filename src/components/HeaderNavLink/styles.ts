import styled from 'styled-components';
import { Paragraph } from 'src/components/Typography';

export const HeaderLinkItem = styled.li`
    padding: 0;
    margin: ${({
        theme: {
            spacing: { small },
        },
    }) => `0 ${small} 0 0`};
`;

export const HoverIndicator = styled.div`
    display: none;
    height: 3px;
    background-color: ${({
        theme: {
            colors: { pink },
        },
    }) => pink};
    margin-top: ${({
        theme: {
            spacing: { tiny },
        },
    }) => tiny};
    width: 100%;
    max-width: 0;
    transition: max-width ease 0.3s;

    max-width: ${({ selected }: { selected: boolean }) =>
        selected ? `150px` : 0};

    @media screen and (min-width: ${({
            theme: {
                breakpoints: { small },
            },
        }) => small}) {
        display: block;
    }
`;

export const RowCenter = styled.div`
    display: flex;
    flex-direction: row;
    justify-content: center;
`;

export const NavLinkStyled = styled.a`
    text-decoration: none;
    display: flex;
    flex-direction: column;
    justify-content: center;
    ${Paragraph} {
        transition: color ease 0.3s, padding 0.3s ease, border 0.3s ease;
        font-weight: 600;
        font-size: ${({
            theme: {
                fontSizes: { extraExtraLarge },
            },
        }) => extraExtraLarge};
        @media screen and (min-width: ${({
                theme: {
                    breakpoints: { small },
                },
            }) => small}) {
            font-size: ${({
                theme: {
                    fontSizes: { medium },
                },
            }) => medium};
        }
    }
    cursor: pointer;

    &.selected {
        ${Paragraph} {
            border-left: ${({
                theme: {
                    colors: { pink },
                },
            }) => `solid 0.25rem ${pink}`};
            padding-left: 0.25em;

            /* we have underlines to indicate selected for large screen width so we don't need to change text color */
            @media screen and (min-width: ${({
                    theme: {
                        breakpoints: { small },
                    },
                }) => small}) {
                border-left: none;
                padding-left: 0;
            }
        }
    }

    &:hover,
    &:focus {
        outline: none;
        ${Paragraph} {
            color: ${({
                theme: {
                    colors: { pink },
                },
            }) => pink};
        }
    }
`;
