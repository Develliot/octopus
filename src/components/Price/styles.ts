import styled from 'styled-components';
import { Span } from 'src/components/Typography';

export const Pounds = styled(Span)`
    display: inline-block;
    font-size: ${({
        theme: {
            fontSizes: { extraExtraExtraLarge },
        },
    }) => extraExtraExtraLarge};
    line-height: ${({
        theme: {
            fontSizes: { extraExtraExtraLarge },
        },
    }) => `calc(${extraExtraExtraLarge} - 0.25em)`};
`;

export const Pence = styled(Span)`
    display: inline-block;
    font-size: ${({
        theme: {
            fontSizes: { large },
        },
    }) => large};
    line-height: ${({
        theme: {
            fontSizes: { large },
        },
    }) => large};
    vertical-align: top;
`;
