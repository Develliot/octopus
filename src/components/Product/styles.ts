import styled from 'styled-components';
import { Span } from 'src/components/Typography';

export const CoverColor = styled.div`
    background-color: ${({
        theme: {
            colors: { darkBlue },
        },
    }) => darkBlue};
    width: 100%;
    height: 300px;
    position: relative;

    @media screen and (min-width: ${({
            theme: {
                breakpoints: { small },
            },
        }) => small}) {
        height: 400px;
    }
`;

export const ImageWrapper = styled.div`
    height: 300px;
    width: 300px;
    overflow: hidden;
    position: absolute;
    bottom: -40px;
`;

export const ImageContainer = styled.img`
    height: 300px;
    width: 300px;
`;

export const Pounds = styled(Span)`
    display: inline-block;
    font-size: ${({
        theme: {
            fontSizes: { extraExtraExtraLarge },
        },
    }) => extraExtraExtraLarge};
    line-height: ${({
        theme: {
            fontSizes: { extraExtraExtraLarge },
        },
    }) => `calc(${extraExtraExtraLarge} - 0.25em)`};
`;

export const Pence = styled(Span)`
    display: inline-block;
    font-size: ${({
        theme: {
            fontSizes: { large },
        },
    }) => large};
    line-height: ${({
        theme: {
            fontSizes: { large },
        },
    }) => large};
    vertical-align: top;
`;

