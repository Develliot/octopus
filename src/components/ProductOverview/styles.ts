import styled from 'styled-components';
import { H3 } from 'src/components/Typography';

export const OverviewImage = styled.img`
    width: 3em;
    height: 3rem;
    margin-right: 1rem;
`;
export const OverviewWrapper = styled.div`
    padding: 1em;
`;

export const OverviewText = styled(H3)`
    transition: color 0.3s ease;
    &:hover {
        color: ${({
            theme: {
                colors: { pink },
            },
        }) => pink};
    }
`;

export const OverviewLink = styled.a`
    text-decoration: none;
    cursor: pointer;
`;
