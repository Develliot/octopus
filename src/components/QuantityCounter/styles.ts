import styled from 'styled-components';
import { Paragraph, Span } from 'src/components/Typography';

export const Counter = styled.div`
    min-width: 100px;
`;

export const CounterLabel = styled(Paragraph)`
    max-width: 100%;
    text-align: center;
`;

export const CounterButton = styled.button`
    height: 2rem;
    width: 2rem;
    border-radius: 0.4rem;
    font-weight: 700;
    border: none;
    font-size: ${({
        theme: {
            fontSizes: { extraExtraLarge },
        },
    }) => extraExtraLarge};
    text-align: center;
    line-height: 1.9rem;
    /* adjust for wonky font */
    padding: 0 0 1.5rem 0.09rem;

    transition: color 0.3s ease, background-color 0.3s ease;
    background-color: ${({
        theme: {
            colors: { blueGrey },
        },
    }) => blueGrey};
    color: ${({
        theme: {
            colors: { white },
        },
    }) => white};
    background-color: ${({
        theme: {
            colors: { blueGrey },
        },
    }) => blueGrey};
    color: ${({
        theme: {
            colors: { white },
        },
    }) => white};

    &:focus,
    &:hover {
        background-color: ${({
            theme: {
                colors: { white },
            },
        }) => white};
        color: ${({
            theme: {
                colors: { blueGrey },
            },
        }) => blueGrey};
    }
`;

export const CounterValue = styled(Span)`
    padding: 0 0.5em;
    font-size: ${({
        theme: {
            fontSizes: { extraExtraExtraLarge },
        },
    }) => extraExtraExtraLarge};
    line-height: ${({
        theme: {
            fontSizes: { extraExtraExtraLarge },
        },
    }) => extraExtraExtraLarge};
`;
