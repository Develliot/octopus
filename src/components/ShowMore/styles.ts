import styled from 'styled-components';
// import { colors } from 'src/theme';

export const ShowMoreText = styled.div`
    position: relative;
    overflow: hidden;
    height: ${({
        isOpen,
        closeHeight,
    }: {
        isOpen: boolean;
        closeHeight: string;
    }) => (isOpen ? `auto` : closeHeight)};

    &.is-open {
        height: auto;
    }

    &::after {
        content: '';
        display: ${({ isOpen }: { isOpen: boolean }) =>
            isOpen ? `none` : `block`};
        position: absolute;
        bottom: 0;
        width: 100%;
        height: 2.5em;
        background: ${({
            color,
            theme: {
                colors,
                colors: { blue },
            },
        }) => `linear-gradient(
            transparent,
            ${color ? colors[color] : blue}
        )`};
    }
`;

export const ShowMoreButton = styled.button`
    background-color: transparent;
    border: none;
    color: ${({
        theme: {
            colors: { pink },
        },
    }) => pink};
    text-decoration: underline;
    padding: 0;

    cursor: pointer;
`;
