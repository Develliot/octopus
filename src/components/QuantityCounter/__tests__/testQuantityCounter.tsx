import React from 'react';
import { shallow, ShallowWrapper } from 'enzyme';
import { QuantityCounter } from '../';
import { CounterButton, CounterValue } from '../styles';

// not much point testing not visible because snapshot
// won't capture changes in styled component
describe('QuantityCounter', () => {
    it('renders correctly', () => {
        const mockOnChangeHandler = jest.fn();
        const wrapper: ShallowWrapper = shallow(
            <QuantityCounter minValue={1} onChange={mockOnChangeHandler} />
        );
        expect(wrapper).toMatchSnapshot();
    });

    it('when value changes onChangeHandlerFires with the correct value', () => {
        const mockOnChangeHandler = jest.fn();
        const wrapper: ShallowWrapper = shallow(
            <QuantityCounter minValue={1} onChange={mockOnChangeHandler} />
        );

        wrapper
            .find(CounterButton)
            .at(1)
            .simulate('click');

        expect(wrapper.find(CounterValue).text()).toEqual('2');
        expect(mockOnChangeHandler).toHaveBeenCalledWith(2);

        wrapper
            .find(CounterButton)
            .at(0)
            .simulate('click');
        expect(wrapper.find(CounterValue).text()).toEqual('1');
        expect(mockOnChangeHandler).toHaveBeenCalledWith(1);
    });

    it('wont change the value below the minimum threshold', () => {
        const mockOnChangeHandler = jest.fn();
        const wrapper: ShallowWrapper = shallow(
            <QuantityCounter minValue={1} onChange={mockOnChangeHandler} />
        );

        wrapper
            .find(CounterButton)
            .at(0)
            .simulate('click');
        expect(wrapper.find(CounterValue).text()).toEqual('1');
        expect(mockOnChangeHandler).toHaveBeenCalledWith(1);
    });
});
