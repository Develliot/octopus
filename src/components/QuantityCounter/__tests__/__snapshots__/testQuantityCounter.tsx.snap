// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`QuantityCounter renders correctly 1`] = `
<styles__Counter>
  <styles__CounterLabel
    bold={true}
    color="blueGrey"
  >
    QTY
  </styles__CounterLabel>
  <Layout__RowCenter>
    <styles__CounterButton
      aria-label="decrement item count"
      onClick={[Function]}
      title="decrement item count"
    >
      -
    </styles__CounterButton>
    <styles__CounterValue
      aria-live="assertive"
      bold={true}
    >
      1
    </styles__CounterValue>
    <styles__CounterButton
      aria-label="increment item count"
      onClick={[Function]}
      title="increment item count"
    >
      +
    </styles__CounterButton>
  </Layout__RowCenter>
</styles__Counter>
`;
